src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -196,15 +196,10 @@ public ParticipationResource(ParticipationService participationService, Programm
 
         ProgrammingExerciseStudentParticipation participation = programmingExerciseParticipationService.findStudentParticipationByExerciseIdAndStudentId(exerciseId,
                 principal.getName());
-        if (participation == null) {
```

Comment:
This part was never reachable before since we checked in the method above is optional is empty and throw an exception.

Reply:
please revert and use the new version of develop

Reply:
Done